code/modules/projectiles/gun.dm

```diff
@@ -636,7 +633,13 @@
 	return launched
 
 /obj/item/weapon/gun/proc/play_fire_sound(var/mob/user, var/obj/item/projectile/P)
-	var/shot_sound = (istype(P) && P.fire_sound)? P.fire_sound : fire_sound
+	var/shot_sound = fire_sound
+
+	if(!shot_sound && istype(P) && P.fire_sound) // If the gun didn't have a fire_sound, but the projectile does...
+		shot_sound = P.fire_sound
+	else

```

Comment:
else catches:
> if the gun has a fire sound
> if the projectile doesn't exist
> if the projectile doesn't have a fire sound

You probably want an
> if(!shot_sound)
> > return

Reply:
Yeah that's probably a bit better.